test/github-issues/4096/issue-4096.ts

```diff
@@ -0,0 +1,61 @@
+import "reflect-metadata";
+import {createTestingConnections, closeTestingConnections, reloadTestingDatabases} from "../../utils/test-utils";
+import {Connection} from "../../../src/connection/Connection";
+import {expect} from "chai";
+import {User} from "./entity/User";
+
+describe("github issues > #4096 SQLite support for orUpdate", () => {
+    let connections: Connection[];
+
+    before(async () => connections = await createTestingConnections({
+        entities: [User],
+        enabledDrivers: ["sqlite"],
+        schemaCreate: true,
+        dropSchema: true,
+    }));
+
+    beforeEach(() => reloadTestingDatabases(connections));
+
+    after(() => closeTestingConnections(connections));
+
+    const user1 = new User();
+    user1.email = "example@example.org";
+    user1.username = "example";
+    user1.bio = "My bio";
+
+    const user2 = new User();
+    user2.email = "example@example.org";
+    user2.username = "example";
+    user2.bio = "Updated bio";
+
+    it("should overwrite using current value in SQLite", () => Promise.all(connections.map(async connection => {
+      try {
+        const UserRepository = connection.manager.getRepository(User);
+
+        await UserRepository
+          .createQueryBuilder()
+          .insert()
+          .into(User)
+          .values(user1)
+          .execute();
+
+        await UserRepository
+          .createQueryBuilder()
+          .insert()
+          .into(User)
+          .values(user2)
+          .orUpdate({
+            conflict_target: [ "email", "username" ],
+            overwrite: ["bio"],
+          })
+          .execute();
+
+        const users = await UserRepository.find();
+        expect(users).not.to.be.undefined;
+        expect(users).to.have.lengthOf(1);
+        expect(users[0]).to.includes({ bio: "Updated bio" });
+      } catch (err) {
+        throw new Error(err);
```

Comment:
Catching errors just to throw them anyway isn't the best practice. Why not just remove whole try ... catch?

Reply:
Totally agree.
I just copied the structure in this tests https://github.com/typeorm/typeorm/pull/3056/files#diff-44f5f56ba7aa275b0e0ed9f63a771c3e just in case I was missing something.